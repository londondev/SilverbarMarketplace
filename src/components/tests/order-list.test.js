import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import  OrderList  from '../order-summary';

const mockOrders=[
  {orderId:1, userId:1, quantity:200, unitPrice:100, orderType:'BUY'},
  {orderId:2, userId:2, quantity:200, unitPrice:150, orderType:'BUY'},
  {orderId:3, userId:1, quantity:300, unitPrice:100, orderType:'BUY'},
]
const defaultProps = {
  orderSummary: mockOrders
};

describe('Order List', () => {

  it('render without crashing', () => {
    shallow(<OrderList {...defaultProps} />);
  });

  it('should render orders', () => {
    const renderedItem = shallow(<OrderList {...defaultProps}  />);
    expect(renderedItem.find('tr')).toHaveLength(5);
  });

  it('should match the snapshot', () =>{
    const renderedItem = renderer.
                        create(<OrderList {...defaultProps}  />)
                        .toJSON();
    expect(renderedItem).toMatchSnapshot();
  })
});