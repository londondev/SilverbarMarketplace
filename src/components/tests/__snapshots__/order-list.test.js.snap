// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order List should match the snapshot 1`] = `
<div>
  <table>
    <tbody>
      <tr>
        <th
          colSpan="2"
        >
          Order Summary(
          )
        </th>
      </tr>
      <tr>
        <th>
          Unit Price
        </th>
        <th>
          Quantity
        </th>
      </tr>
      <tr>
        <td>
          £
          100
        </td>
        <td>
          200
        </td>
      </tr>
      <tr>
        <td>
          £
          150
        </td>
        <td>
          200
        </td>
      </tr>
      <tr>
        <td>
          £
          100
        </td>
        <td>
          300
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
